code/script.js

```diff
@@ -262,82 +270,136 @@ const selectQuestion = () => {
-  if (group === 'accessories') {
+  const { attribute, category, value } = currentQuestion
+
+  if (category === 'accessories') {
     if (keep) {
       alert(
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
+        `Yes, Yes, Yes the person wears ${attribute}! Keeping all the people that wears ${attribute}.`
       )
     } else {
       alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
+        `Nope, the person doesn't wear ${attribute}! Removing all the people that wears ${attribute}.`
       )
     }
-  } else if (group === 'other') {
-    // Similar to the one above
-  } else {
+  } else if (category === 'hair color') {
+    if (keep) {
+      alert(`Yeeeeey, the person has ${value} hair! Keeping all the people with ${value} hair.`
+      )
+    } else {
+      alert(
+      `Sorry, the person doesn't have ${value} hair. Removing people with ${value} hair.`
+      )
+    }
+  } else if (category === 'eye color') {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+      alert(
+        `Yeeep, the person has ${value} eyes! Keeping all the people with ${value} eyes.`
+      )
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+      alert(
+        `Negative!, the person doesn't have ${value} eyes. Removing people with ${value} eyes.`
+        )
     }
+  } else if (category === 'other') {
+    if (keep) {
+      alert(
+        `Yeep Yeep, the person is a ${attribute}. Keeping all the people that are ${attribute}s.`
+      )
+  } else {
+      alert(
+        `No No No, the person is not a ${attribute}. Removing all the people that are ${attribute}s.`
+      )
+    } 
   }
 
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
 
+  // filter to keep or remove based on the keep variable.
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
+  } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  }
+  generateBoard()
   // Invoke a function to redraw the board with the remaining people.
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
+  const confirmed = confirm(`Are you really sure you want to guess on ${suspect}? Maybe you have got it wrong.`)
+
+  if (confirmed) {
+    checkMyGuess(suspect)
+  }
+// store the interaction from the player in a variable.
   // remember the confirm() ?
   // If the player wants to guess, invoke the checkMyGuess function.
```

Comment:
remember to always clean your code from all console.log()s and commented out code blocks or guides from the teachers 🤓